tests/multi_ptr/multi_ptr_convert_assignment_ops.h

```diff
@@ -0,0 +1,199 @@
+      bool res = false;
+      {
+        sycl::buffer<bool> res_buf(&res, sycl::range<1>(1));
+        sycl::buffer<T> val_buffer(&value, sycl::range<1>(1));
+        queue.submit([&](sycl::handler &cgh) {
+          auto res_acc =
+              res_buf.template get_access<sycl::access_mode::write>(cgh);
+          auto acc_for_mptr =
+              val_buffer.template get_access<sycl::access_mode::read>(cgh);
+
+          if constexpr (src_space ==
+                        sycl::access::address_space::global_space) {
+            auto val_acc =
+                val_buffer.template get_access<sycl::access_mode::read>(cgh);
+            cgh.single_task([=] {
+              const src_multi_ptr_t mptr_in(acc_for_mptr);
+              dst_multi_ptr_t mptr_out;
+
+              mptr_out = mptr_in;
+
+              // Check that second mptr has the same value as first mptr
+              res_acc[0] = *(mptr_out.get_raw()) == acc_for_mptr[0];
+            });
+          } else {
+            sycl::local_accessor<T> local_acc(r, cgh);
+            cgh.parallel_for(
+                sycl::nd_range<1>(r, r), [=](sycl::nd_item<1> item) {
+                  if constexpr (src_space ==
+                                sycl::access::address_space::local_space) {
+                    auto ref = local_acc[0];
+                    value_operations::assign(ref, expected_val);
+
+                    const src_multi_ptr_t mptr_in(acc_for_mptr);
+                    dst_multi_ptr_t mptr_out;
+
+                    mptr_out = mptr_in;
+
+                    // Check that second mptr has the same value as first mptr
+                    res_acc[0] = *(mptr_out.get_raw()) == acc_for_mptr[0];
+                  } else {
+                    T private_val =
+                        user_def_types::get_init_value_helper<T>(expected_val);
+
+                    const src_multi_ptr_t mptr_in = sycl::address_space_cast<
+                        sycl::access::address_space::generic_space, decorated,
+                        T>(&private_val);
+                    dst_multi_ptr_t mptr_out;
+
+                    mptr_out = mptr_in;
+
+                    // Check that second mptr has the same value as first mptr
+                    res_acc[0] = *(mptr_out.get_raw()) == acc_for_mptr[0];
+                  }
+                });
+          }
+        });
+      }
+      CHECK(res);
+    }
+
+    SECTION(
+        section_name(
+            "Check &operator=(multi_ptr<value_type, AS, IsDecorated>&&)")
+            .with("T", type_name)
+            .with("src address_space", src_address_space_name)
+            .with("src decorated", src_is_decorated_name)
+            .with("dst address_space", "access::address_space::generic_space")
+            .with("dst decorated", dst_is_decorated_name)
+            .create()) {
+      bool res = false;
+      {
+        sycl::buffer<bool> res_buf(&res, sycl::range<1>(1));
+        sycl::buffer<T> val_buffer(&value, sycl::range<1>(1));
```

Comment:
```suggestion
        sycl::buffer<bool> res_buf(&res, r);
        sycl::buffer<T> val_buffer(&value, r);
```

Reply:
I've applied this comment [here](https://github.com/KhronosGroup/SYCL-CTS/pull/349/commits/8d52d86ffc87806de55e3915cf575748ddbea725#diff-aaaedf36f568a5a0b02e7455e96e48b9f6c1f06093a504cca944e7edb623965dL123-R123)